code/game/machinery/bots/ed209bot.dm

```diff
@@ -649,11 +653,12 @@ Auto Patrol[]"},
 					name = "taser ED-209 assembly"
 				else
 					return
+			if(!user.unEquip(W))
+				return

```

Comment:
Similar thing here with name, if the check fails the name will still have changed. Easiest solution is probably to use a var/newname = "bluetag stuff" and after the check success do name = newname.